src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsMapping.java

```diff
@@ -0,0 +1,255 @@
+
+    @Value("${exui.url}")
+    private static String exUiUrl;
+
+    private HearingsMapping() {
+
+    }
+
+    public static void updateFlags(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+        caseData.setAutoListFlag(shouldBeAutoListed(caseData.getAutoListFlag()));
+        caseData.setHearingsInWelshFlag(shouldBeHearingsInWelshFlag(caseData.getHearingsInWelshFlag()));
+        caseData.setAdditionalSecurityFlag(shouldBeAdditionalSecurityFlag(caseData.getAdditionalSecurityFlag()));
+        caseData.setSensitiveFlag(shouldBeSensitiveFlag(caseData.getSensitiveFlag()));
+    }
+
+    public static HmcCaseDetails createHmcCaseDetails(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+
+        return HmcCaseDetails.builder()
+                .caseDeepLink(getCaseDeepLink(caseData.getCcdCaseId()))
+                .caseManagementLocationCode(getCaseManagementLocationCode(caseData.getCaseManagementLocation()))
+                .build();
+    }
+
+    public static HearingRequest createHearingRequest(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+
+        HearingRequest hearingRequest = HearingRequest.builder().build();
+
+        hearingRequest.setInitialRequestTimestamp(LocalDateTime.now());
+        hearingRequest.setAutoListFlag(caseData.getAutoListFlag());
+        hearingRequest.setInWelshFlag(caseData.getHearingsInWelshFlag());
+        hearingRequest.setIsLinkedFlag(isCaseLinked(caseData));
+        hearingRequest.setAdditionalSecurityFlag(caseData.getAdditionalSecurityFlag());
+        hearingRequest.setSensitiveFlag(caseData.getSensitiveFlag());
+        hearingRequest.setInterpreterRequiredFlag(
+                isInterpreterRequired(caseData.getAdjournCaseInterpreterRequired()));
+        hearingRequest.setHearingType(getHearingType(caseData));
+        hearingRequest.setFirstDateTimeMustBe(getFirstDateTimeMustBe(caseData));
+        hearingRequest.setHearingWindowRange(getHearingWindowRange(caseData));
+        hearingRequest.setDuration(getHearingDuration(caseData));
+        hearingRequest.setHearingPriorityType(getHearingPriority(
+                caseData.getAdjournCaseCanCaseBeListedRightAway(),
+                caseData.getUrgentCase()));
+        hearingRequest.setHearingLocations(getHearingLocations(caseData.getCaseManagementLocation()));
+        hearingRequest.setFacilitiesRequired(getFacilitiesRequired(caseData));
+        hearingRequest.setListingComments(getListingComments(
+                caseData.getAppeal(), caseData.getOtherParties()));
+        hearingRequest.setLeadJudgeContractType(getLeadJudgeContractType(caseData));
+        hearingRequest.setPanelRequirements(getPanelRequirements(caseData));
+
+        return hearingRequest;
+    }
+
+    public static YesNo shouldBeAutoListed(YesNo autoListFlag) {
+        boolean isYes = YesNo.isYes(autoListFlag);
```

Comment:
Are these local variables needed? Think it would look clearer without.

Reply:
They will be when the method is implemented, this the TODO, I can remove them if you really want?

Reply:
I see.. that's fine leave them in.